docs/index.asciidoc

```diff
@@ -0,0 +1,488 @@
+===== `local_lookups`
+
+  * Value type is <<array,array>>
+  * Default value is `[]`
+
+The array should contain one or more Hashes. Each Hash represents a lookup
+enhancement. Each Hash is validated according to the table below.
+
+[cols="<,<,<",options="header",]
+|=======================================================================
+|Setting |Input type|Required
+| id|string|No
+| query|string|Yes
+| parameters|hash|Yes
+| target|string|No
+| default_hash|hash|No
+| tag_on_failure|string|No
+| tag_on_default_use|string|No
+|=======================================================================
+
+Local_lookups Field Descriptions::
+id:::
+An optional identifier. This is used to identify the lookup that is
+generating error messages and log lines. If you omit this setting then a
+default id is used instead.
+query:::
+A SQL SELECT statement that
+parameters:::
+A key/value Hash or dictionary. The key (LHS) is the text that is
+substituted for in the SQL statement
+`SELECT * FROM sensors WHERE reference = :p1`. The value (RHS)
+is the field name in your event. The plugin reads the value from
+this key out of the event and substitutes that value into the
+statement, e.g. `parameters => { "p1" => "ref" }`. Quoting is
+automatic - you do not need to put quotes in the statement.
+Only use the field interpolation syntax on the RHS if you need to
+add a prefix/suffix or join two event field values together to build
+the substitution value. For example, imagine an IOT message that has
+an id and a location and you have a table of sensors that have a
+column of `id-loc_id`, in this case your parameter hash would look
+like this: `parameters => { "p1" => "%{[id]}-%{[loc_id]}" }`
+target:::
+An optional name for the field that will receive the looked up data.
+If you omit this setting then the `id` setting (or the default id) is
+used. The looked up data, an array of results converted to Hashes, is
+never added to the root of the event. If you want to do this then you
+should use the `add_field` setting to achieve this. This means that
+you are in full control of how the fields/values are put in the root
+of the event, e.g.
+`add_field => { user_firstname => "%{[user][0][firstname]}" }` -
+where `[user]` is the target field `[0]` is the first result in the
+array and `[firstname]` is the key in the result hash.
+default_hash:::
+An optional hash that will be put in the target field array when the
+lookup returns no results. Use this if you need to ensure that later
+references in other parts of the config actually refer to something.
+tag_on_failure:::
+An optional string that overrides the plugin level setting. This is
+useful when defining multiple lookups.
+tag_on_default_use:::
+An optional string that overrides the plugin level setting. This is
+useful when defining multiple lookups.
+
+Further documentation describing this syntax can be found https://github.com/jmettraux/rufus-scheduler#parsing-cronlines-and-time-strings[here].
```

Comment:
This line should go with the scheduler info

Reply:
doh